### ISSUE TITLE
ColumnsWizard - 68/78 Release/Status

### ISSUE THREAD

**Issue by cleidigh** [micz/ColumnsWizard#95]
Latest Beta:  https://github.com/micz/ColumnsWizard/blob/v8.0.0/xpi/columnswizard-8.0.0-b6-tb.xpi

Status: b5 fixes menu removal , shutdown, custom c widths,first cleanup

- [x] Initial beta conversion
- [x] first pass for 78
- [x] post b2 - Windows 78/68 default columns
- [x] post b3 - fix custom columns
- [x] b4 - default, fixes, keep, column menu open
- [x] b5
- [x] b6
- [ ]

**Comment by nkgm-bs** [comment]
Hi,
i've found some issues with b3:

- When other add-ons add custom columns cw shows them with their id not with their label
- When selecting one of your added entries in the column picker menu the menu closes. Normal behaviour ist to stay open.
- When selecting a column in the 'special column' tab of your options, it can't be deselected. The options tab has to be closed and reopened to do this.
- When cw is disabled in the add-on settings all your added columns are still there! It is essential to remove all ui elements (and observers, listeners) added by an add-on, if the add-on is disabled or uninstalled.
- Even with the 'Debug' option unchecked there are many entries from cw in the console log.

And i would recommend that you add a prefix (e.g.'CW:') to the log entries. And/or you should also add a prefix (e.g. 'cw_') to your javascript files. This makes it much easier to recognize, which addon shows the entries. And it made it possible, to filter the log entries.

Regards,
Günter

**Comment by cleidigh** [comment]
@nkgm-bs 
thanks for the testing and feedback!
these are very good points.  the beta is still pretty rough
for the variety of issues moving to 78,  moving columns is still messed up also
question
can you give me an example of one of the other extensions dealing with,columns
where you saw that issue?
Christopher

**Comment by BillH99999** [comment]
I installed beta 4.  In initial testing it seems to work fine for me.  I only use saving and resetting of the default columns as well as using the default columns on new folders.  That seems to work fine.  I'll keep playing around to see what happens.

**Comment by cleidigh** [comment]
@nkgm-bs 
I believe b6 addresses each of the issues you mentioned
All
other than cleanup for if everybody can do a bit of testing
that would be great then I can officially Release

Christopher

**Comment by BillH99999** [comment]
@cleidigh 

I tested save as default, reset to default, and creating a new folder using the default columns and all are still working great.

Thanks!
Bill

**Comment by Ircama** [comment]
columnswizard-8.0.0-b6-tb works well, I have not found problems, thanks a lot for your work

**Comment by nkgm-bs** [comment]
There is still the problem when selecting a column in the 'special column' tab it can't be deselected. The options tab has to be closed and reopened to do this.

The 'New...' button in the  'special column' tab does nothing. The error console shows
  Uncaught ReferenceError: miczColumnsWizardPrefPage2 is not defined
     onclick chrome://columnswizard/content/settings.html:1

Other things i mentioned earlier seems to be ok.

**Comment by cleidigh** [comment]
@nkgm-bs 
thanks for the ongoing feedback. I cannot reproduce so far the de-selection problem
I see I broke the new button
What platform,/version are you using?
Christopher

**Comment by nkgm-bs** [comment]
@cleidigh
Platform is Windows 10/TB 78.5
I've checked the de-selection problem with different profiles and with other add-ons deactivated. The problem still happens.

Thereby i've found another problem. When i dis(!)able my add-on ShowInOut (which also adds custom columns) the  'special column' tab is completely empty and error console shows
`Uncaught (in promise) TypeError: win3pane.document.getElementById(...) is null
    getColLocalizedString chrome://columnswizard/content/mzcw-defaultcolslist.jsm:616
    updateDefaultColsList chrome://columnswizard/content/mzcw-defaultcolslist.jsm:145
    initDefaultColsList chrome://columnswizard/content/mzcw-defaultcolslist.jsm:111
`
Seems that you try to access columns which no longer exists.

**Comment by cleidigh** [comment]
@nkgm-bs 
my dictation software inserted 'page2' obviously my last commit,oh well
have to figure out what's different for you on the checkboxes
Christopher

**Comment by cleidigh** [comment]
I need to see if I need an update observer for other columns

**Comment by nkgm-bs** [comment]
This problem will arise very seldom i think. Probably its enough to wrap the code in a try/catch block.

**Comment by cleidigh** [comment]
I'm going to install the extension to better understand the interaction
I tested two other extensions which add columns but I am sure
with all the permutations (includes my limitations)
it's easy to miss things
question
when you are unable to un check a custom column is one you created?

**Comment by cleidigh** [comment]
okay I think I know what you see
if you uncheck a custom, col on the custom tab it toggles there
but it remains on the new columns list
is that accurate?

**Comment by nkgm-bs** [comment]
> when you are unable to un check a custom column is one you created?

No, it's one of the predefined (e.g. cc). Currently i can't test with a new column because 'New...' is not working.

> if you uncheck a custom, col on the custom tab it toggles there
but it remains on the new columns list
is that accurate?

Ah, its 'Custom columns'. No, if i check the predefined 'cc' in error and want to immediately uncheck it, i can't. The checkboxes for 'searchable' works as expected.

**Comment by nkgm-bs** [comment]
I've started over. Deinstalled cw, removed all cw settings from prefs.js, installed cw again.

When i open the 'custom columns', the searchable checkboxes for the items 'contentbase', 'replyto', 'xoriginalfrom', 'xspamscore' are all checked! Deselecting all four items (which at this time works) and activating (e.g.) 'contentbase' et voilá all four searchable checkboxes are checked again. And can't be deselected!

**Comment by IdahoGuy** [comment]
Windows 10, TB 78.6.1
Right click drop down menu on a message includes 'CW Header Edit >>' but nothing happens when I click on it and the secondary drop down menu does not show. I really hoped this was the answer I want at the moment!

**Comment by BlueWings76** [comment]
Hello Christopher,

Today I installed columnswizard-8.0.0-b6-tb.xpi into Thunderbird 78.13.0 (32-bit) (latest autoupdated) under Windows 7 Ultimate 64 bit.
Only the save and reset columns to/as CW default functions work, nothing else.
There's a ColumnsWizard button in the toolbar, but it's grayed out, and clicking it produces the following in the error console :

Uncaught TypeError: tab.browser is undefined
    openSettingsTab chrome://columnswizard/content/mzcw-overlay.js:624
    openSettingsTab chrome://columnswizard/content/mzcw-overlay.js:623
    oncommand chrome://messenger/content/messenger.xhtml:1
mzcw-overlay.js:624:8
    openSettingsTab chrome://columnswizard/content/mzcw-overlay.js:624
    forEach self-hosted:225
    openSettingsTab chrome://columnswizard/content/mzcw-overlay.js:623
    oncommand chrome://messenger/content/messenger.xhtml:1

Am I doing something wrong ?

Thanks, Peter

**Comment by JLuc** [comment]
I installed V8 B6 on TB 78 on Ubuntu 21.04. 
UI is sometimes a bit tricky but as for now it now works fine : i needed a new "reply to" column, and i got it now !

**Comment by jmdaemon** [comment]
What's the status on columnswizard-8.0.0-b6 ? Thunderbird says the extension is not compatible with Thunderbird v91.2.0. As well the last time I tried it, none of the functionality was able to work, and I couldn't define, use, or edit any columns.

**Comment by stephankoenig** [comment]
I there any way to add a "Junk Status +" Column as JunQuilla had it before?

Is that existing information?

I am willing to donate to support the development.
